docs/sdk/src/reference_docs/frame_runtime_upgrades_and_migrations.rs

```diff
@@ -0,0 +1,135 @@
+//! # Runtime Upgrades
+//!
+//! At their core, blockchain logic consists of
+//!
+//! 1. on-chain state and
+//! 2. a state transition function
+//!
+//! In Substrate-based blockchains, state transition functions are referred to as
+//! [runtimes](https://docs.substrate.io/learn/runtime-development/).
+//!
+//! Traditionally, before Substrate, upgrading state transition functions required node
+//! operators to download new software and restart their nodes in a process called
+//! [forking](https://en.wikipedia.org/wiki/Fork_(blockchain)).
+//!
+//! Substrate-based blockchains do not require forking, and instead upgrade runtimes
+//! in a process called "Runtime Upgrades".
+//!
+//! Forkless runtime upgrades are a defining feature of the Substrate framework. Updating the
+//! runtime logic without forking the code base enables your blockchain to seemlessly evolve
+//! over time in a deterministic, rules-based manner. It also removes ambiguity for node operators
+//! and other participants in the network about what is the canonical runtime.
+//!
+//! This capability is possible due to the runtime of a blockchain existing in on-chain storage.
+//!
+//! ## Performing a Runtime Upgrade
+//!
+//! To upgrade a runtime, an [`Origin`](frame_system::RawOrigin) with the necesarry permissions
+//! (usually via governance) executes [`set_code`] (or [`set_code_without_checks`]) with the
+//! desired new blob.
```

Comment:
Noteworthy that this is a common example, yet anything changing `:code:` is a runtime upgrade. In fact, in reality, usually the origin will `utility::schedule` a `system::set_code`.